src/main/java/io/cdap/plugin/sfmc/source/MarketingCloudClient.java

```diff
@@ -140,6 +141,9 @@ public MarketingCloudObjectInfo fetchObjectSchema(SourceObject object)
       response = PaginationETSoapObject.continueRequest(client, null, response.getRequestId(),
                                                         filter);
       rows.addAll(response.getObjects());
+      if (isPreviewEnabled) {
```

Comment:
Why are you bailing out if preview is enabled? 
If the issue is that preview is taking too long because it's reading too many records, we should fix [CDAP-18647](https://cdap.atlassian.net/browse/CDAP-18647). Looks like there's an open PR for it - https://github.com/cdapio/cdap/pull/13806

Reply:
> Why are you bailing out if preview is enabled?
> If the issue is that preview is taking too long because it's reading too many records, we should fix [CDAP-18647](https://cdap.atlassian.net/browse/CDAP-18647). Looks like there's an open PR for it - [cdapio/cdap#13806](https://github.com/cdapio/cdap/pull/13806)

Preview is taking too long because we are fetching all the record while preview is enabled even though if we set the preview limit on cdap side. so here we are fetching only one batch data while preview is Enabled.